clang/lib/CConv/Constraints.cpp

```diff
@@ -81,6 +98,14 @@ bool Constraints::addConstraint(Constraint *C) {
   // Check if C is already in the set of constraints. 
   if (constraints.find(C) == constraints.end()) {
     constraints.insert(C);
+
+    if (Geq *G = dyn_cast<Geq>(C)) {
+      if (G->constraintIsChecked())
+        ChkCG->addConstraint(G, *this);
+      else
+        PtrTypCG->addConstraint(G, *this);
+    }
```

Comment:
What happens with implication constraints? It is confusing to me that you "eagerly" add constraints to the graph when they are arrows, but (necessarily) wait until later for the implication constraints. How do you ensure everything is consistent? For example, what if a Geq constraint `X` triggers an Implication constraint during solving, which adds a Geq. But if `X` is removed later?

Reply:
You are right. I changed the code so that we create the constraint graph during solving.